x/msgfee/model.go

```diff
@@ -0,0 +1,75 @@
+package msgfee
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ orm.CloneableData = (*MsgFee)(nil)
+
+func (mf *MsgFee) Validate() error {
+	if mf.MsgPath == "" {
+		return errors.Wrap(errors.ErrInvalidModel, "invalid message path")
+	}
+	if mf.Fee.IsZero() {
+		return errors.Wrap(errors.ErrInvalidModel, "invalid fee")
+	}
+	if err := mf.Fee.Validate(); err != nil {
+		return errors.Wrap(err, "invalid fee")
+	}
+	return nil
+}
+
+func (mf *MsgFee) Copy() orm.CloneableData {
+	return &MsgFee{
+		MsgPath: mf.MsgPath,
+		Fee:     *mf.Fee.Clone(),
+	}
+}
+
+type MsgFeeBucket struct {
+	orm.Bucket
+}
+
+// NewMsgFeeBucket returns a bucket for keeping track of fees for eeach message
+// type. Message fees are indexed by the corresponding message path.
+func NewMsgFeeBucket() *MsgFeeBucket {
+	b := orm.NewBucket("msgfee", orm.NewSimpleObj(nil, &MsgFee{}))
+	return &MsgFeeBucket{
+		Bucket: b,
+	}
+}
+
+// Create adds given message fee instance to the store.
+func (b *MsgFeeBucket) Create(db weave.KVStore, mf *MsgFee) (orm.Object, error) {
+	key := []byte(mf.MsgPath)
```

Comment:
Fair enough, use the path as key, and also as part of the value struct.
I tried not to duplicate this info before in eg. Ticker, but it was at the cost of much mental bookkeeping later on, I think this is the simpler and cleaner approach.